### PR TITLE
Add attendance list without tests 

### LINKED ISSUE #66: Create list of attendance for v1.3
* Create attendance list class
* Allow CRUD for Attendance

### PR DESCRIPTION
**Pull request by Evande1**
- attendance add 1 c/mod s/10 (Max size of 12) 
- attendance mark 1 l/1 m/1 (0 or 1, else reject)
- attendance delete 1 
- closes #66

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AttendanceAddCommand.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            + " " + COMMAND_WORD + " 1 "
+            + PREFIX_CLASS_GROUP + "CS2103T " + PREFIX_SIZE + "10";
+
+    private final Index index;
+    private final String mod;
+    private final String size;
+
+    /**
+     * Creates an AttendanceAddCommand to add to the specified {@code Student}
+     * @param index of the person in the filtered person list to edit the remark
```
Comment (gowribhat) [inline src/main/java/seedu/address/logic/commands/AttendanceAddCommand.java:43]: Needs an empty line between the description and parameter section.

File: src/main/java/seedu/address/logic/commands/AttendanceMarkCommand.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            + "Example: " + ATTENDANCE_COMMAND_WORD
+            + " " + COMMAND_WORD + " 1 "
+            + PREFIX_LESSON + "3 " + PREFIX_MARK + "1";
+    private final Index studentIndex;
+    private final String lessonIndex;
+    private final Attendance attendance;
+
+    /**
+     * Creates an AttendanceMarkCommand to mark attendance of the specified {@code Student}
+     * @param studentIndex
```
Comment (gowribhat) [inline src/main/java/seedu/address/logic/commands/AttendanceMarkCommand.java:40]: Add line.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -83,6 +84,21 @@ public static Attendance parseAttendance(Optional<String> attendance) throws Par
         }
         return new Attendance(trimmedAttendance);
     }
+    /**
+     * Parses {@code String size}
+     * @param size with the input
```
Comment (gowribhat) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:89]: Add line.

File: src/main/java/seedu/address/model/attendance/Attendance.java [hunk 4]
```diff
@@ -5,16 +5,15 @@
...
 public class Attendance {
     public static final String VALIDATION_REGEX = "[0-1]";
     public static final String MESSAGE_CONSTRAINTS = "attendance is either a 0(absent) or 1(present)";
     public final String value;
 
     /**
-     * Create attendance object
-     * @param attendance
+     * Constructs an {@code Email}.
+     * @param attendance A valid attendance.
```
Comment (gowribhat) [inline src/main/java/seedu/address/model/attendance/Attendance.java:16]: Add line.

File: src/main/java/seedu/address/model/attendance/AttendanceList.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+     */
+    public AttendanceList() {
+        this.isEmpty = true;
+        this.mod = "NA";
+        this.attendanceList = new ArrayList<>();
+    }
+
+    /**
+     * Constructs an {@code AttendanceList}.
+     * @param mod A module of the attendance list
```
Comment (gowribhat) [inline src/main/java/seedu/address/model/attendance/AttendanceList.java:51]: Add line.

File: src/main/java/seedu/address/model/attendance/AttendanceList.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    /**
+     * Constructs an {@code AttendanceList}.
+     * @param mod A module of the attendance list.
+     * @param size The size of the attendance list
+     */
+    public AttendanceList(String mod, String size) {
+        requireNonNull(mod, size);
+        checkArgument(isValidSize(size), MESSAGE_CONSTRAINTS);
+        this.isEmpty = Integer.parseInt(size) == 0;
+        this.mod = isEmpty ? "NA" : mod;
```
Comment (gowribhat) [inline src/main/java/seedu/address/model/attendance/AttendanceList.java:33]: Could "NA" be abstracted out?

File: src/test/java/seedu/address/logic/commands/AttendanceCommandTest.java [hunk 7]
```diff
@@ -1,115 +1,117 @@
...
-        assertFalse(standardCommand.equals(new AttendanceCommand(INDEX_SECOND_STUDENT,
-                new Attendance(VALID_ATTENDANCE_AMY))));
-
-        // different remark -> returns false
-        assertFalse(standardCommand.equals(new AttendanceCommand(INDEX_FIRST_STUDENT,
-                new Attendance(VALID_ATTENDANCE_BOB))));
-    }
-
-}
+//package seedu.address.logic.commands;
```
Comment (thamruicong) [inline src/test/java/seedu/address/logic/commands/AttendanceCommandTest.java:1]: Test cases to be done

File: src/test/java/seedu/address/logic/commands/AttendanceDeleteCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package seedu.address.logic.commands;
```
Comment (thamruicong) [inline src/test/java/seedu/address/logic/commands/AttendanceDeleteCommandTest.java:1]: Test cases to be done

File: src/test/java/seedu/address/logic/commands/AttendanceMarkCommandTest.java [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package seedu.address.logic.commands;
```
Comment (thamruicong) [inline src/test/java/seedu/address/logic/commands/AttendanceMarkCommandTest.java:1]: Test cases to be done

File: src/test/java/seedu/address/logic/parser/AttendanceCommandParserTest.java [hunk 10]
```diff
@@ -1,44 +1,44 @@
...
-        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, AttendanceCommand.MESSAGE_USAGE);
-
-        // no parameters
-        assertParseFailure(parser, AttendanceCommand.COMMAND_WORD, expectedMessage);
-
-        // no index
-        assertParseFailure(parser, AttendanceCommand.COMMAND_WORD + " " + nonEmptyAttendance, expectedMessage);
-    }
-}
+//package seedu.address.logic.parser;
```
Comment (thamruicong) [inline src/test/java/seedu/address/logic/parser/AttendanceCommandParserTest.java:1]: Test cases to be done

File: src/main/java/seedu/address/logic/commands/AttendanceDeleteCommand.java [hunk 11]
```diff
@@ -10,54 +10,48 @@
...
+            + " " + COMMAND_WORD + " 1 ";
 
     private final Index index;
-    private final Attendance attendance;
 
     /**
-     * @param index of the person in the filtered person list to edit the remark
-     * @param attendance of the person to be updated to
+     * Creates an AttendanceDeleteCommand to delete the attendance of the specified {@code Student}
+     * @param index of the student to delete the attendance list from
```
Comment (gowribhat) [inline src/main/java/seedu/address/logic/commands/AttendanceDeleteCommand.java:36]: Add line

File: src/main/java/seedu/address/model/attendance/AttendanceList.java [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+    //if size is 0, empty list, elif up to 12, else reject.
+    public static final String VALIDATION_REGEX = "[0-9]|1[012]";
+    public static final String MESSAGE_CONSTRAINTS = "Invalid size";
+    private final String mod;
+    private List<Attendance> attendanceList;
+    private final boolean isEmpty;
+
+    /**
+     * Constructs an {@code AttendanceList}.
+     * @param mod A module of the attendance list.
```
Comment (gowribhat) [inline src/main/java/seedu/address/model/attendance/AttendanceList.java:26]: Add line

**Review by gowribhat — COMMENTED** [review]
Minor coding standard violations, otherwise LGTM 👍🏽

File: src/main/java/seedu/address/logic/parser/AttendanceCommandParser.java [hunk 13]
```diff
@@ -1,37 +1,59 @@
...
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.HelpCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
-//import seedu.address.model.attendance.Attendance;
+
 
 
 /**
- * Parses input arguments and creates a new {@code AttendanceCommand} object
+ * Parses input arguments and creates new AttendanceParsers object
```
Comment (zzthian) [inline src/main/java/seedu/address/logic/parser/AttendanceCommandParser.java:18]: AttendanceParsers object? Not sure what you mean here. AttendanceCommand object seems fine

File: src/main/java/seedu/address/logic/commands/AttendanceAddCommand.java [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+        return new CommandResult(generateSuccessMessage(editedStudent));
+    }
+
+    /**
+     * Generates a command execution success message based on whether
+     * the attendance is added to or removed from
+     * {@code studentToEdit}.
+     */
+    private String generateSuccessMessage(Student studentToEdit) {
+        String message = MESSAGE_ADD_ATTENDANCE_SUCCESS;
```
Comment (zzthian) [inline src/main/java/seedu/address/logic/commands/AttendanceAddCommand.java:84]: Need not create a message variable imo. Just whack the whole MESSAGE_ADD_ATTENDANCE_SUCCESS in.

**Review by zzthian — APPROVED** [review]
LGTM I believe we can merge it.

**Review by leewaichong — APPROVED** [review]
Great Implementation of the Attendance List!

**Pull request merged**